tests/test_composite.py

```diff
@@ -320,22 +341,42 @@ def test_multi_slice_index(ant, sphere, uniform, airplane, globe):
     # Now check everything
     sub = multi[0:3]
     assert len(sub) == 3
-    for i in range(3):
-        assert id(sub[i]) == id(multi[i])
+    for i in range(len(sub)):
+        assert sub[i] is multi[i]
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1]
-    assert len(sub) == len(multi) == multi.n_blocks
-    for i in range(multi.n_blocks):
-        assert id(sub[i]) == id(multi[i])
+    assert len(sub) + 1 == len(multi)
+    for i in range(len(sub)):
+        assert sub[i] is multi[i]
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1:2]
-    assert len(sub) == 3
-    for i in range(3):
+    assert len(sub) == 2
+    for i in range(len(sub)):
         j = i*2
-        assert id(sub[i]) == id(multi[j])
+        assert sub[i] is multi[j]
         assert sub.get_block_name(i) == multi.get_block_name(j)
 
 
+def test_slice_defaults(ant, sphere, uniform, airplane, globe):
+    multi = multi_from_datasets(ant, sphere, uniform, airplane, globe)
+    assert multi[:] == multi[0:len(multi)]
+
+
+def test_slice_negatives(ant, sphere, uniform, airplane, globe):
+    multi = multi_from_datasets(ant, sphere, uniform, airplane, globe)
+    test_multi = pyvista.MultiBlock({key: multi[key] for key in multi.keys()[::-1]})
+    assert multi[::-1] == test_multi
+
+    test_multi = pyvista.MultiBlock({key: multi[key] for key in multi.keys()[-2:]})
+    assert multi[-2:] == test_multi
+
+    test_multi = pyvista.MultiBlock({key: multi[key] for key in multi.keys()[:-1]})
+    assert multi[:-1] == test_multi
+
+    test_multi = pyvista.MultiBlock({key: multi[key] for key in multi.keys()[-1:-4:-2]})
+    assert multi[-1:-4:-2] == test_multi
```

Comment:
I was going to remark that it would be more readable to define a list of datasets and slice those in the dict comprehension, but I then realised that we need the keys too. Enumerating the sliced list and calling `multi.get_block_name()` would be equally convoluted. Too bad.

Reply:
In this case, I think the awkwardness highlights the confusing state of the current API.  There is no obviously simple way to reverse a MultiBlock.

Reply:
There is now: `multi[::-1]` ;)